template-only-docs/template-technical-design-principles.md

```diff
@@ -0,0 +1,48 @@
+# Template technical design principles
+
+This document outlines the technical design principles that the template-infra project follows. These principles guide the development of the infrastructure template.
+
+## Design for simplicity
+
+### Design for the common case
+
+- Optimize for the common case
+- Avoid adding unnecessary flexibility that increases complexity
+- Ensure that uncommon cases can be addressed through explicit customization
+
+### Foster developer understanding and ownership
+
+- Ensure the infrastructure is transparent and comprehensible to project teams.
+- Encourage developers to understand the inner workings of the infrastructure so they are equipped to extend or customize it as needed.
+
+### Maintain a level of complexity comparable to what project teams would manage if building the functionality themselves
+
+- Avoid adding complexity to the project that arises from the fact that the code came from a template
+- Avoid adding complexity to the project that arises from the need to support multiple use cases
+
+### Establish strong conventions and sensible defaults
+
+- Use conventions to eliminate unnecessary decision-making and prevent bikeshedding. For example, require a `Makefile` with a `release-build` target in all applications rather than allow project teams to decide what command to use for building the container.
```

Comment:
I love this one. Some of the other contractors on grants.gov (outside Nava) have different opinions here, but I personally love the consistency